h2/src/main/org/h2/engine/Database.java

```diff
@@ -1521,12 +1521,12 @@ private synchronized void closeOpenFilesAndUnlock(boolean flush) {
             reconnectModified(false);
             if (store != null && store.getMvStore() != null && !store.getMvStore().isClosed()) {
                 long maxCompactTime = dbSettings.maxCompactTime;
-                if (compactMode == CommandInterface.SHUTDOWN_COMPACT) {
-                    store.compactFile(dbSettings.maxCompactTime);
-                } else if (compactMode == CommandInterface.SHUTDOWN_DEFRAG) {
-                    maxCompactTime = Long.MAX_VALUE;
-                } else if (getSettings().defragAlways) {
+                if (compactMode == CommandInterface.SHUTDOWN_COMPACT ||
+                        compactMode == CommandInterface.SHUTDOWN_DEFRAG ||
+                        getSettings().defragAlways) {
                     maxCompactTime = Long.MAX_VALUE;
+                } else {
```

Comment:
SHUTDOWN DEFRAG should probably be the same as just SHUTDOWN since it's original use in PageStore just does a light re-ordering of pages.
Also I note that we don't support SHUTDOWN IMMEDIATELY properly here, where maxCompactTime should be zero.

Reply:
IMMEDIATELY goes a different code path, no changes here. As fa as DEFRAG, I’d rather make it synonym of COMPACT, because that’s how it sounds.

Reply:
Cool. I don't have strong feelings about this.